tests/helpers/setup-unhandled-exception-listener.js

```diff
@@ -0,0 +1,44 @@
+import QUnit from 'qunit';
```

Comment:
The existing `assert-errors.js` and this new test helper seem addressing the same problem. Both are even taking the same approach of overwriting `QUnit.onUncaughtException`. Main difference seems to be that yours also adds an event handler for `unhandledrejection`, which prevents the default error handling. Beside that the public API of both is different as well.

Did you tried using the existing one? Maybe extending it with the `unhandledrejection` event listener.

I fear having to rest helpers for the same issue complicates the code base unnecessarily.

Reply:
I did not notice the existing one. Strangely enough, there seems to be a difference between `Qunit.onUncaughtException` and `Qunit.onUnhandledRejection`. The second did not seem to work in this case, so I updated the code in `assert-error.js` to use that. It did not seem to influence the other usage of the helper.

Reply:
I done a code search for the existing test helper. It seems to be not used at all.

It's imported in `tests/integration/components/bs-button-test.js`. But all tests using it are currently skipped. We have an open ticket about it: #1615 Let's try activating them again after landing the improved test helper from you.